app/code/Bulbulatory/Recomendations/Model/Recommendation.php

```diff
@@ -1,24 +1,91 @@
 <?php
 namespace Bulbulatory\Recomendations\Model;
 
-class Recommendation extends \Magento\Framework\Model\AbstractModel implements \Magento\Framework\DataObject\IdentityInterface
+use Magento\Framework\Model\AbstractExtensibleModel;
+use Bulbulatory\Recomendations\Api\Data\RecommendationInterface;
+
+class Recommendation extends AbstractExtensibleModel implements RecommendationInterface
 {
-    const CACHE_TAG = 'bulbulatory_recomendations_recommendation';
-    protected $_cacheTag = 'bulbulatory_recomendations_recommendation';
-    protected $_eventPrefix = 'bulbulatory_recomendations_recommendation';
+    const ID = 'recommendation_id';
+    const CUSTOMER_ID = 'customer_id';
+    const EMAIL = 'email';
+    CONST HASH = 'hash';
+    CONST STATUS = 'status';
+    CONST CREATION_DATE = 'create_at';
+    CONST CONFIRMATION_DATE = 'confirmed_at';
 
     protected function _construct()
     {
         $this->_init(\Bulbulatory\Recomendations\Model\ResourceModel\Recommendation::class);
     }
 
-    public function getIdentities()
+    public function getId()
+    {
+        return $this->_getData(self::ID);
+    }
+
+    public function setId($id)
+    {
+        return $this->setData(self::ID, $id);
+    }
+
+    public function getCustomerId()
+    {
+        return $this->_getData(self::CUSOMER_ID);
+    }
+
+    public function setCustomerId($id)
+    {
+        return $this->setData(self::CUSOMER_ID, $id);
```

Comment:
Literówka :stuck_out_tongue: 
```suggestion
        return $this->setData(self::CUSTOMER_ID, $id);
```